src/dali_executor/dali_executor.cc

```diff
@@ -50,21 +49,20 @@ template <bool owns>
 std::vector<shape_and_type_t>
 DaliExecutor::Run(const std::vector<IODescr<owns>>& inputs)
 {
-  auto& pipeline =
-      pipeline_pool_.Get(serialized_pipeline_, max_batch_size_, device_id_);
-  SetupInputs(pipeline, inputs);
+  SetupInputs(inputs);
   try {
-    pipeline.Run();
-    pipeline.Output();
+    pipeline_.Run();
+    pipeline_.Output();
   }
   catch (std::runtime_error& e) {
-    pipeline_pool_.Remove(serialized_pipeline_, max_batch_size_, device_id_);
+    pipeline_ = DaliPipeline(serialized_pipeline_, max_batch_size_,
+                             num_threads_, device_id_);
```

Comment:
Maybe it would be worth to consider allowing `DaliPipeline` to reset itself? `serialized_pipeline_` shouldn't change after reset, so something like `pipeline_.Reset()` should suffice

Reply:
done